File: Python/ByRound/2067/2067_C_Devyatkino.py
# Contest 2067, Problem C: Devyatkino
# URL: https://codeforces.com/contest/2067/problem/C

for i in [*open(0)][1:]:
    t = int(i)
    l = []
    n = t
    for i in range(1, 10):
        c = 0
        while "7" not in str(t):
            t += int(i * "9")
            c += 1
        l += [c]
        t = n
    print(min(l))

File: CPP/ByTier/C/2067_C_Devyatkino.cpp
/*
 * Problem URL : https://codeforces.com/problemset/problem/2067/C
 * Submit Date : 2025-08-23
 */

#include <bits/stdc++.h>

using namespace std;

int _;

int main() {
  scanf("%d", &_);
  for (; _--;) {
    int n;
    scanf("%d", &n);
    for (int i = 0; i <= 9; i++) {
      int k = n - i, mi = 7;
      for (; k > 0; mi = min(mi, (17 - k % 10) % 10), k /= 10)
        ;
      if (mi <= i) {
        printf("%d\n", i);
        break;
      }
    }
  }
}